src/plugins/intel_gpu/src/graph/impls/onednn/gemm_onednn.cpp

```diff
@@ -138,56 +138,84 @@ struct gemm_onednn : typed_primitive_onednn_impl<gemm> {
             return false;
         };
 
-        if (batched_dims_can_be_removed) {
-            if (has_transpose_order(prim->input0_transpose_order)) {
+        // Check whether transpose_order has transpose only to the last two elements.
+        // Retur true when transpose_order is 0, 1, 3, 2.
+        auto has_transpose_order_xy_only = [](std::vector<int64_t> transpose_order) {
+            for (size_t i = 0; i < transpose_order.size(); i++) {
+                if (i < transpose_order.size() - 2) {
+                    if (i != static_cast<size_t>(transpose_order[i])) {
+                        return false;
+                    }
+                } else if (i == transpose_order.size() - 2) {   // last -1
+                    if (i != static_cast<size_t>(transpose_order[transpose_order.size() - 1])) {
+                        return false;
+                    }
+                } else {    // last
+                    if (i != static_cast<size_t>(transpose_order[transpose_order.size() - 2])) {
+                        return false;
+                    }
+                }
+            }
```

Comment:
* isn't it necessary to check whether it is 4d or not? what happens to 5d case?

* what about 
```
            for (size_t i = 0; i < transpose_order.size() - 2; i++) {
                    if (i != static_cast<size_t>(transpose_order[i])) {
                        return false;
                    }
                }
             size_t last_idx = transpose_order.size() - 1;
             if (transpose_order[last_idx] != last_idx - 1)
                 return false;
             if (transpose_order[last_idx - 1] != last_idx)
                 return false;
```

Reply:
Good idea, much graceful!
We don't need to check whether 4d or 5d in here.  Actual onednn size of dims will be up to 4, this function will be used `ab` `abc` `abcd` Other cases are handled by `transpose_dims_and_format_tag`

There's no possibility for `Can't find fusable transpose format` in static, Dynamic can be because of all transposes fusion in ngraph without gemm whitelist but low possibility and will follow up it future.